_posts/2019-09-02-shorter-isn't-better.md

```diff
@@ -0,0 +1,250 @@
+---
+layout: post
+title: "Shorter functions are't always better"
+date: 2019-09-05 10:00:00 +0800
+categories: blog
+author: Lex Huang
+---
+
+# Shorter functions aren't always better
+
+This is just a thought that came to me during day to day code reviews and I want to share the idea with you. My point is that for functions, readability, understandability and maintainability is much more important the length and the abstraction level.
+
+## A Trivial Example
+
+Consider you want to implement a simple algorism that show the balance in foreign currency given a list of incomes and outcomes. I will do this in `Javascript`. For the sake of simplicity, I will **ignore** the `float point issue` of ECMAScript and `type/value validation` for the arguments.
```

Comment:
Consider ```that``` you want to ... in foreign currency```,``` given a list of incomes and ```expenditures```. (The word outcome has a completely different meaning.). For the sake of simplicity, let's use `Javascript` and **ignore** the `floating point issue` and `type/value validation`.